main_ISTA.py

```diff
@@ -60,6 +60,8 @@ def __init__(self, data: Dataset, num_subsets: int = 7, step_size: float = 0.1,
                                                                     initial_image=data.OSEM_image)
         for f in obj_funs: # add prior to every objective function
             f.set_prior(data.prior)
+        data.prior.set_penalisation_factor(data.prior.get_penalisation_factor() / len(obj_funs))
```

Comment:
I'd suggest putting these lines before the `set_prior`. I guess it doesn't matter as it's a reference, but it probably is less confusing to most people. Also add a comment on why we do this (prior is evenly distributed over all subsets)